packages/core/src/Synchronizer.ts

```diff
@@ -306,17 +309,17 @@ export class Synchronizer extends EventEmitter {
     }
 
     protected async _checkCurrentEpoch(epoch: number) {
-        const currentEpoch = await this.loadCurrentEpoch()
+        const currentEpoch = this.calcCurrentEpoch()
         if (epoch !== Number(currentEpoch)) {
             throw new Error(
                 `Synchronizer: Epoch (${epoch}) must be the same as the current epoch ${currentEpoch}`
             )
         }
     }
 
-    protected async _checkValidEpoch(epoch: number) {
-        const currentEpoch = await this.loadCurrentEpoch()
-        if (epoch > Number(currentEpoch)) {
+    protected async _checkValidEpoch(epoch: bigint | number) {
+        const currentEpoch = this.calcCurrentEpoch()
+        if (BigInt(epoch) > currentEpoch) {
```

Comment:
There is a
https://github.com/Unirep/Unirep/blob/f15c0cb0290dd1817487d6312b922a0e2019a73a/packages/core/src/Synchronizer.ts#L612
in `attestationEvent(event: ethers.Event, db: TransactionDB)`
I think if `_checkCurrentEpoch()` means to get the epoch on-chain (not the db stored epoch)
then the `await this._checkCurrentEpoch(_epoch)` can be changed to `await this._checkValidEpoch(_epoch)` or we should also keep the `readCurrentEpoch` to read the current db processed epoch.

Or after epoch ends, the attestation event will not be processed correctly (the synchronizer will break)

Reply:
Right, added this logic back in here: [52cfb42](https://github.com/Unirep/Unirep/pull/165/commits/52cfb42b99f7b16a2cc88c7b89bfcf933b9113ec)

The `_checkCurrentEpoch` function is only used in a single place. I removed it because the name is ambiguous.